src/Microsoft.OData.Edm/Csdl/Serialization/EdmModelCsdlSchemaWriter.cs

```diff
@@ -400,6 +401,31 @@ internal void WriteOperationParameterElementHeader(IEdmOperationParameter parame
             }
         }
 
+        internal void WriteOperationParameterEndElement(IEdmOperationParameter parameter)
+        {
+            IEdmOptionalParameter optionalParameter = parameter as IEdmOptionalParameter;
+            if (optionalParameter != null && !(optionalParameter.VocabularyAnnotations(this.model).Where(a => a.Term == CoreVocabularyModel.OptionalParameterTerm).Count() > 0))
+            {
+                string defaultValue = optionalParameter.DefaultValueString;
+                List<EdmPropertyConstructor> properties = new List<EdmPropertyConstructor>();
+                var optionalValue = new EdmRecordExpression(properties);

```

Comment:
I also tend to avoid using var; I'll make this an EdmRecordExpression.

---
In reply to: [123292911](https://github.com/OData/odata.net/pull/828#discussion_r123292911) [](ancestors = 123292911,123061452)